tools/src/main/java/org/apache/kafka/tools/reassign/ReassignPartitionsCommand.java

```diff
@@ -0,0 +1,1523 @@
+                                                                            List<String> topics
+    ) throws ExecutionException, InterruptedException {
+
+        Map<TopicPartition, List<Integer>> res = new HashMap<>();
+        describeTopics(adminClient, new HashSet<>(topics)).forEach((topicName, topicDescription) ->
+            topicDescription.partitions().forEach(info -> res.put(
+                new TopicPartition(topicName, info.partition()),
+                info.replicas().stream().map(Node::id).collect(Collectors.toList())
+            )
+        ));
+        return res;
+    }
+
+    /**
+     * Get the current replica assignments for some partitions.
+     *
+     * @param adminClient     The AdminClient to use.
+     * @param partitions      The partitions to get information about.
+     * @return                A map from partitions to broker assignments.
+     *                        If any topic can't be found, an exception will be thrown.
+     */
+    static Map<TopicPartition, List<Integer>> getReplicaAssignmentForPartitions(Admin adminClient,
+                                                                                Set<TopicPartition> partitions
+    ) throws ExecutionException, InterruptedException {
+        Map<TopicPartition, List<Integer>> res = new HashMap<>();
+        describeTopics(adminClient, partitions.stream().map(TopicPartition::topic).collect(Collectors.toSet())).forEach((topicName, topicDescription) -> {
+            topicDescription.partitions().forEach(info -> {
+                TopicPartition tp = new TopicPartition(topicName, info.partition());
+                if (partitions.contains(tp))
+                    res.put(tp, info.replicas().stream().map(Node::id).collect(Collectors.toList()));
+            });
+        });
+        return res;
+    }
+
+    /**
+     * Find the rack information for some brokers.
+     *
+     * @param adminClient         The AdminClient object.
+     * @param brokers             The brokers to gather metadata about.
+     * @param enableRackAwareness True if we should return rack information, and throw an
+     *                            exception if it is inconsistent.
+     *
+     * @return                    The metadata for each broker that was found.
+     *                            Brokers that were not found will be omitted.
+     */
+    static List<BrokerMetadata> getBrokerMetadata(Admin adminClient, List<Integer> brokers, boolean enableRackAwareness) throws ExecutionException, InterruptedException {
+        Set<Integer> brokerSet = new HashSet<>(brokers);
+        List<BrokerMetadata> results = adminClient.describeCluster().nodes().get().stream()
+            .filter(node -> brokerSet.contains(node.id()))
+            .map(node -> (enableRackAwareness && node.rack() != null)
+                ? new BrokerMetadata(node.id(), Optional.of(node.rack()))
+                : new BrokerMetadata(node.id(), Optional.empty())).collect(Collectors.toList());
```

Comment:
Please move .collect to new line to align with other Stream methods.

Reply:
Done.